static/js/typeahead_helper.js

```diff
@@ -261,9 +261,13 @@ export function sort_people_for_relevance(objs, current_stream_name, current_top
 }
 
 export function compare_by_popularity(lang_a, lang_b) {
-    const diff = pygments_data.langs[lang_b].priority - pygments_data.langs[lang_a].priority;
-    if (diff !== 0) {
-        return diff;
+    const lang_a_data = pygments_data.langs[lang_a];
+    const lang_b_data = pygments_data.langs[lang_b];
+    if (lang_a_data && lang_b_data) {
+        const diff = lang_b_data.priority - lang_a_data.priority;
+        if (diff !== 0) {
+            return diff;
+        }
     }
     return util.strcmp(lang_a, lang_b);
```

Comment:
I misread the fallthrough behavior here, which got me thinking about whether this function could be clearer. 

I think the best structure might actually be to extract a function `get_language_priority(lang)`, and have that function return a fixed value for languages that don't exist in `pygments_data` (probably with a comment explaining that decision). And then we could unconditionally compare by priority and then by string in all cases.

One significant benefit of that change is that the comparison function would be obviously sound, in the sense that A>B and B>C implies A>C. I actually suspect that the version you've written doesn't have that property, because comparisons between two languages where only one has pygments data are alphabetical. Might be good to include a test case for that class of corner case. (I haven't looked at whether we sort using this function with an algorithm that would be sad with that detail).

Reply:
Ah good feedback. Thanks.

I added couple commits to address this. I opted for not introducing popularity scores for languages that don't have it, so to avoid having any kind of assumption about popularity ranges.